### ISSUE TITLE
[16.3.1] The matrix factorization model의 loss(식 16.3.3)의 regularizing term

### ISSUE THREAD

**Issue by staycozyboy** [woowacampers/d2l-pytorch#8]
![image](https://user-images.githubusercontent.com/37537248/108787364-ef0d7180-75b8-11eb-8079-322319876eb5.png)

>The regularizing term  $λ(∥P∥^2_F+∥Q∥^2_F+b^2_u+b^2_i)$  is used to avoid over-fitting by penalizing the magnitude of the parameters.

over-fitting을 방지하기 위해 파라미터의 크기에 패널티를 부과한다는 것이 직관적으로 잘 와닿지는 않아서 올려봅니다. <br>
제가 이해한 건

1. 모델 내에서 파라미터가 미치는 영향력을 감소시켜 일반화 성능을 높이겠다 이런 말같고, 
2. loss에 loss와 상관없는 term을 넣어줘도 학습이 적당히 잘되는 것은 예상되는 loss 값에 비해 상대적으로 regularizing term의 값이 충분히 작다는 보장이 있어서 그런건가..? 싶습니다.

**Comment by jiwon-ryu** [comment]
저도 이거 궁금했어요!
regularizing, weight decay가 overfitting을 방지하는 원리가 궁금합니다

**Comment by staycozyboy** [comment]
### 스터디
지원님: 람다 뒤의 텀이 사이즈를 loss로 주는 것. 사이즈를 줄이는 방향으로 학습을 시키는 것 같다.

[https://daeson.tistory.com/184](https://daeson.tistory.com/184)

파라미터의 사이즈를 작게 만들어주면, 고차원 계수의 항을 거의 0에 가깝게 날릴 수 있기 때문에, overfitting 방지.

한결님: over-fitting이 심하게 표현력이 좋은것인데, 어쨌든 고차항의 계수에 패널티를 많이 줘서 표현력을 제한해줌.

- L2 normalization 찾아보시면 도움이 될 것 같습니다.

**Comment by 1gyeol-KIM** [comment]
overfitting을 완화시키기 위해 기존의 cost function에서 regularization term (식 16.3.3의 lambda term)을 추가시켰습니다.
구체적으로 regularization이 어떻게 overfit을 방지하는지는 오늘 노션의 학습콘텐츠 자료 참고하면 좋을 것 같습니다 ㅎㅎ
그리고 train data가 아닌 real data에서 좋은 성능을 내기 위해 bias와 variance간에는  tradeoff는 반드시 생길 수 밖에 없으며 이를 **"bias-variance trade-off"** 라고 부른다고 합니다.
bias와 variance설명과 regularization 관계를 comment에 남기려다.. 좋은 reference들을 발견해서 첨부해 드립니다!
참고하시면 좋을 것 같아요
***References
[bias vs variance](https://modulabs-biomedical.github.io/Bias_vs_Variance)
[Lasso, Ridge regularization - 회귀에서 selection과 shrinkage](https://sosal.kr/1104)
